docs/configuration/overview/overview.rst

```diff
@@ -57,6 +61,24 @@ stats_flush_interval_ms
   performance reasons Envoy latches counters and only flushes counters and gauges at a periodic
   interval. If not specified the default is 5000ms (5 seconds).
 
+watchdog_miss_timeout_ms
+  *(optional, integer)* The time in milliseconds after which we count a nonresponsive thread in the
+  "server.watchdog_miss" statistic. If not specified the default is 200ms.
+
+watchdog_megamiss_timeout_ms
+  *(optional, integer)* The time in milliseconds after which we count a nonresponsive thread in the
+  "server.watchdog_mega_miss" statistic. If not specified the default is 1000ms.
+
+watchdog_kill_timeout_ms
```

Comment:
I think leaving the kill behavior off by default is OK for now.  It's easy to change the defaults later or just use a custom config.